analysis_options.yaml

```diff
@@ -0,0 +1,63 @@
+analyzer:
+  strong-mode: true
+
+linter:
+  rules:
```

Comment:
Here's the list we used in another library, maybe we should discuss and come up with an informal standard at least...

```
linter:
  rules:
    # Errors
    - avoid_empty_else
    - hash_and_equals
    - iterable_contains_unrelated_type
    - test_types_in_equals
    - throw_in_finally
    - unrelated_type_equality_checks
    # Style
    - always_declare_return_types
    - annotate_overrides
    - avoid_init_to_null
    - avoid_return_types_on_setters
    - await_only_futures
    - camel_case_types
    - constant_identifier_names
    - empty_constructor_bodies
    - implementation_imports
    - library_names
    - library_prefixes
    - non_constant_identifier_names
    - one_member_abstracts
    - overridden_fields
    - package_api_docs
    - package_prefixed_library_names
    - prefer_is_not_empty
    - slash_for_doc_comments
    - sort_constructors_first
    - sort_unnamed_constructors_first
    - super_goes_last
    - type_init_formals
    - unnecessary_getters_setters
    # Pub rules
    - package_names
```